src/wp-includes/post.php

```diff
@@ -6076,11 +6086,13 @@ function get_pages( $args = array() ) {
 	 * Filters the retrieved list of pages.
 	 *
 	 * @since 2.1.0
+	 * @since 6.3.0 The `$query_args` parameter was added.
 	 *
 	 * @param WP_Post[] $pages       Array of page objects.
 	 * @param array     $parsed_args Array of get_pages() arguments.
+	 * @param array     $query_args  Array of arguments passed to WP_Query.
 	 */
-	return apply_filters( 'get_pages', $pages, $parsed_args );
+	return apply_filters( 'get_pages', $pages, $parsed_args, $query_args );
```

Comment:
Why is the new parameter here needed? That seems separate from the actual new filter this PR adds. What is the rationale for adding `$query_args` here?

Reply:
It helps if you are doing something with the `get_pages` filter to have this context.

Reply:
Doing what? I think as long as we don't have any real use-case, we shouldn't add parameters to filters, just because it could help. I don't see anyone in the ticket asking for this either.

Reply:
With this new context variable, you could easily detect if your the site is running WP 6.3+. Making using the filter much easier. Adding extra parameter to the filter is nice to have, I can revert, I don't see how it hurts.

Reply:
You can also detect that through other ways that aren't more complicated. 

While it doesn't really "hurt", adding another parameter means another thing to maintain, to consider in the future in case we want to change something e.g. around backward compatibility. For example, what if later we realize another parameter makes more sense on this filter than this one? We would have already added something this parameter, so then we would need to add yet another parameter, and at some point it just becomes messy because of too many parameters. We shouldn't add parameters just because we can, but only when we have established there is a reasonable need for it.

Reply:
> You can also detect that through other ways that aren't more complicated.

Now that `$query_args` can be filtered, it may not be easy to get these args passed to WP_Query. I think this context is useful here.

Reply:
You can also get those args by using the `pre_get_posts` action from the `WP_Query` instantiated above.

So again, on the one hand, you're still bringing up theoretical reasons why this could be useful, without any concrete example. We can always introduce it later if a concrete example comes along. But we can't remove it once it's there, so it's safer to not come to too quick conclusions.

Another point related to that is that, _if_ we want to add a new parameter, we could argue that it should be the `WP_Query` instance instead. That would provide access to more context than just the `$query_args`. So this type of discussion is also something we would need to have when introducing a parameter like this, but at this point I find that rather pointless since we have no good example for it.

Reply:
Okay, I will remove. I have no strong feeling about this one.